File: convert_icon.js
/**
 * Convert SVG icon to PNG for Flutter launcher icons
 * Requires: npm install sharp
 */

const fs = require('fs');
const path = require('path');

async function convertIcon() {
  try {
    // Check if sharp is installed
    let sharp;
    try {
      sharp = require('sharp');
    } catch (e) {
      console.log('Error: sharp package not installed.');
      console.log('Please install it: npm install sharp');
      console.log('\nAlternatively, use an online converter:');
      console.log('1. Go to https://convertio.co/svg-png/');
      console.log('2. Upload assets/icon/secure_bdrs_icon.svg');
      console.log('3. Set size to 1024x1024');
      console.log('4. Download and save as assets/icon/secure_bdrs_icon.png');
      process.exit(1);
    }

    const svgPath = path.join(__dirname, 'assets', 'icon', 'secure_bdrs_icon.svg');
    const pngPath = path.join(__dirname, 'assets', 'icon', 'secure_bdrs_icon.png');

    if (!fs.existsSync(svgPath)) {
      console.log(`Error: ${svgPath} not found!`);
      process.exit(1);
    }

    // Convert SVG to PNG
    await sharp(svgPath)
      .resize(1024, 1024)
      .png()
      .toFile(pngPath);

    console.log('✓ Successfully converted SVG to PNG!');
    console.log(`  Input: ${svgPath}`);
    console.log(`  Output: ${pngPath}`);
    console.log(`  Size: 1024x1024 pixels\n`);
    console.log('Next step: Run "flutter pub run flutter_launcher_icons"');

  } catch (error) {
    console.error('Error:', error.message);
    console.log('\nAlternative: Use an online SVG to PNG converter');
    process.exit(1);
  }
}

convertIcon();

File: convert_icon.py
"""
Script to convert SVG icon to PNG for Flutter launcher icons
Requires: pip install cairosvg pillow
"""

try:
    import cairosvg
    from PIL import Image
    import io
    import os
    
    # Convert SVG to PNG
    svg_path = "assets/icon/secure_bdrs_icon.svg"
    png_path = "assets/icon/secure_bdrs_icon.png"
    
    if not os.path.exists(svg_path):
        print(f"Error: {svg_path} not found!")
        exit(1)
    
    # Convert SVG to PNG (1024x1024 for app icons)
    png_data = cairosvg.svg2png(url=svg_path, output_width=1024, output_height=1024)
    
    # Save PNG file
    with open(png_path, 'wb') as f:
        f.write(png_data)
    
    print(f"✓ Successfully converted {svg_path} to {png_path}")
    print(f"  Size: 1024x1024 pixels")
    
except ImportError:
    print("Error: Required packages not installed.")
    print("Please install: pip install cairosvg pillow")
    print("\nAlternatively, you can:")
    print("1. Use an online SVG to PNG converter")
    print("2. Open the SVG in a browser and export as PNG (1024x1024)")
    print("3. Use Inkscape or another vector graphics tool")
except Exception as e:
    print(f"Error: {e}")
    print("\nAlternative: Use an online converter or graphics tool")

